Update emulation available on iOS (#2167)
OWASP/owasp-mstg: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

The goal of emulator detection is to increase the difficulty of running the app on an emulated device. This forces the reverse engineer to defeat the emulator checks or utilize the physical device, thereby barring the access required for large-scale device analysis.

> Previously XCode was the only choice if a developer needed to mimic an iOS or iPadOS device. Mimic really is the best term to describe the level of emualtion that is really possible within XCode, hence emulator protection was not so much a concern regarding iOS apps in contrast to Android with a wide range of emulation choices available. With the release of [Corellium](https://www.corellium.com/), a publicly available iOS emulator, the playfield has shifted towards an emulation, still limited (i.e. IPAs need resigning), but as a typical SaaS offering, employing a vast number of devices is just a few mouse clicks away with the limiting factor just being available funds.

Reviewer: Mimic really is the best term to describe the level of **emualtion** that is really possible within XCode
emulation <- spelling error.

Author: Thanks for reviewing, fixed typo.

With Apple M1 and M2 hardware widely available, traditional checks for presence of x86 / x64 architecture might not suffice. A different strategy is to  know the features and limitations available with the offering to successfully defend against a device emulation. With i.e. Corellium not offering support for iCloud, cellular services, camera, NFC, bluetooth, Appstore access or GPU hardware emulation ([Metal](https://developer.apple.com/documentation/metal/gpu_devices_and_work_submission/getting_the_default_gpu) unavailable) smartly combining checks involving any unavailable features are one indicator for the presence of an emulated environment.

> Pairing these results with the ones from 3rd party frameworks such as [iOS Security Suite](https://github.com/securing/IOSSecuritySuite#emulator-detector-module) (Open Source), [Trusteer](https://www.ibm.com/products/trusteer-mobile-sdk/details) or a No-Code solution such as [Appdome](https://www.appdome.com/) will provide a good line of defense against attacks utilizing emulators. 

Reviewer 2: ```suggestion
Pairing these results with the ones from 3rd party frameworks such as [iOS Security Suite](https://github.com/securing/IOSSecuritySuite#emulator-detector-module), [Trusteer](https://www.ibm.com/products/trusteer-mobile-sdk/details) https://www.ibm.com/products/trusteer-mobile-sdk/details or a no-code solution such as [Appdome](https://www.appdome.com/) (commercial solution) will provide a good line of defense against attacks utilizing emulators. 
```

The goal of emulator detection is to increase the difficulty of running the app on an emulated device. This forces the reverse engineer to defeat the emulator checks or utilize the physical device, thereby barring the access required for large-scale device analysis.

> As discussed in the section [Testing on the iOS Simulator](0x06b-Basic-Security-Testing.md "Testing on the iOS Simulator") in the basic security testing chapter, the only available simulator is the one that ships with Xcode. Simulator binaries are compiled to x86 code instead of ARM code and apps compiled for a real device (ARM architecture) don't run in the simulator, hence _simulation_ protection was not so much a concern regarding iOS apps in contrast to Android with a wide range of _emulation_ choices available. 

Reviewer 2: ```suggestion
As discussed in the section [Testing on the iOS Simulator](0x06b-Basic-Security-Testing.md "Testing on the iOS Simulator") in the basic security testing chapter, the only available simulator is the one that ships with Xcode. Simulator binaries are compiled to x86 code instead of ARM code and apps compiled for a real device (ARM architecture) don't run in the simulator, hence _simulation_ protection was not so much a concern regarding iOS apps in contrast to Android with a wide range of _emulation_ choices available.
```

However, since its release, [Corellium](https://www.corellium.com/) (commercial tool) has enabled real emulation, [setting itself apart from the iOS simulator](https://www.corellium.com/compare/ios-simulator). In addition to that, being a SaaS solution, Corellium enables large-scale device analysis with the limiting factor just being available funds.

With Apple Silicon (ARM) hardware widely available, traditional checks for the presence of x86 / x64 architecture might not suffice. One potential detection strategy is to identify features and limitations available for commonly used emulation solutions. For instance, Corellium doesn't support iCloud, cellular services, camera, NFC, Bluetooth, App Store access or GPU hardware emulation ([Metal](https://developer.apple.com/documentation/metal/gpu_devices_and_work_submission/getting_the_default_gpu)). Therefore, smartly combining checks involving any of these features could be an indicator for the presence of an emulated environment.

> Pairing these results with the ones from 3rd party frameworks such as [iOS Security Suite](https://github.com/securing/IOSSecuritySuite#emulator-detector-module), [Trusteer](https://www.ibm.com/products/trusteer-mobile-sdk/details) https://www.ibm.com/products/trusteer-mobile-sdk/details or a no-code solution such as [Appdome](https://www.appdome.com/) (commercial solution) will provide a good line of defense against attacks utilizing emulators. 

Reviewer 2: ```suggestion
Pairing these results with the ones from 3rd party frameworks such as [iOS Security Suite](https://github.com/securing/IOSSecuritySuite#emulator-detector-module), [Trusteer](https://www.ibm.com/products/trusteer-mobile-sdk/details) or a no-code solution such as [Appdome](https://www.appdome.com/) (commercial solution) will provide a good line of defense against attacks utilizing emulators.
```
